src/compat/db_json.cpp

```diff
@@ -1126,6 +1148,33 @@ db_json_get_string_from_value (const JSON_VALUE *doc, bool copy)
     }
 }
 
+STATIC_INLINE const char *const
+db_json_bool_to_string (bool b)
+{
+  return b ? "true" : "false";
+}
+
+const char *
+db_json_get_bool_as_str_from_value (const JSON_VALUE *doc, bool copy)
+{
+  if (doc == NULL)
+    {
+      assert (false);
+      return NULL;
+    }
+
+  assert (db_json_get_type_of_value (doc) == DB_JSON_BOOL);
+
+  if (copy)
```

Comment:
1. Please use separate functions to copy and to get a string. Returning copy as const char * (and then stripping its const property) is not advisable. Please also correct db_json_get_string_from_value, it has the same wrong behavior.

2. db_json_bool_to_string is not safe. It returns stack memory, but ends up being stored in a DB_VALUE (and stripped of its const property). It may cause memory corruptions. I think we have to allocate the boolean string, I don't know another safe way.